proto/pulumi/provider.proto

```diff
@@ -59,7 +59,24 @@ service ResourceProvider {
     // GetSchema fetches the schema for this resource provider.
     rpc GetSchema(GetSchemaRequest) returns (GetSchemaResponse) {}
 
-    // CheckConfig validates the configuration for this resource provider.
+    // `CheckConfig` validates a set of configuration inputs that will be passed to this provider instance.
```

Comment:
I would love to have a dedicated devdocs page on configuring providers that talks about cross-request flow scenarios. We could link it here. That would cover default vs explicit, has prior state vs no prior state, what's in the state, cascading replacements, etc.

Reply:
So my plan is, when these are all done, to port over the implementer's guide and make it (hopefully) a lot richer, linking to these references but (again, hopefully) giving a more tutorial-like feeling of how this all stitches together to make a provider. Additionally, as mentioned elsewhere -- lots of this stuff is just "how it is now"; I'd really like to flesh it out as we uncover the edge cases in the engine/introduce better ways of doing it like `Handshake`, etc.